ori/testing/volume/fake.go

```diff
@@ -0,0 +1,152 @@
+// Copyright 2022 OnMetal authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package volume
+
+import (
+	"context"
+	"crypto/rand"
+	"encoding/hex"
+	"strconv"
+	"sync"
+	"time"
+
+	ori "github.com/onmetal/onmetal-api/ori/apis/volume/v1alpha1"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+func filterInLabels(labelSelector, lbls map[string]string) bool {
+	return labels.SelectorFromSet(labelSelector).Matches(labels.Set(lbls))
+}
+
+const defaultIDLength = 63
+
+func generateID(length int) string {
+	data := make([]byte, (length/2)+1)
+	for {
+		_, _ = rand.Read(data)
+		id := hex.EncodeToString(data)
+
+		// Truncated versions of the id should not be numerical.
+		if _, err := strconv.ParseInt(id[:12], 10, 64); err != nil {
+			continue
+		}
+
+		return id[:length]
+	}
+}
```

Comment:
We should use `idgen.Default` for this and not copy / paste the (in this case redundant) implementation from the fake machine runtime.